In [42]:
import pandas as pd
import numpy as np
from datetime import date, datetime
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import GridSearchCV
from sklearn.svm import SVC

In [43]:
df = pd.read_csv('cleaned_df.csv')
df.drop('Unnamed: 0', axis=1, inplace=True)
pd.set_option('display.max_columns', None)

In [44]:
#Doing the datetime conversions and creating an Age column.

datetime_cols = ['DOB','DateofHire','DateofTermination','LastPerformanceReview_Date']

for i in datetime_cols:
    df[i] = pd.to_datetime(df[i])

df['Age'] = (datetime.today().year - df['DOB'].dt.year)

In [45]:
#Here I select the variables that I think will be most important in deciding if an employee will be terminated.

split_df = df[['Salary','Department','PerformanceScore','EngagementSurvey','EmpSatisfaction','SpecialProjectsCount','DaysLateLast30', 'Absences','Age','Termd']]

In [46]:
split_df = pd.get_dummies(split_df, columns=['Department','PerformanceScore'], dtype=int)

In [47]:
scaler = StandardScaler()

X = split_df.drop('Termd', axis=1)
y = split_df['Termd']

X = scaler.fit_transform(X)

In [48]:
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42, test_size=0.2)

In [49]:
def modelaccuracy(model):
    model.fit(X_train, y_train)
    result = accuracy_score(y_test, (model.predict(X_test)))
    print(f'Model Accuracy is {result}')

In [50]:
clf = SVC(random_state=42)

modelaccuracy(clf)

Model Accuracy is 0.7258064516129032


In [51]:
clf_param = {'kernel':['linear', 'poly', 'rbf'], 'degree':[2, 3, 4, 5, 6]}

grid_clf = GridSearchCV(clf, clf_param)

modelaccuracy(grid_clf)

Model Accuracy is 0.7258064516129032


- Model accuracy is 0.73%. I used a scaler to normalize the data and ran the model with GridSearchCV to find the best hyperparameters.
- I used SVC mainly because this dataset is extremely small and this model is good for small datasets.